docs/packages/dev-workflow.md

```diff
@@ -0,0 +1,86 @@
+# Dev workflow
+
+This document provides guidance on how to build and publish package and repo bundles and test them on your local cluster.
+
+1. Build the container images for all the packages
+
+   ```shell
+      OCI_REGISTRY=${REGISTRY} make docker-all
+   ```
+
+   This would build, push a docker image for the component and also replaces the newImage path in the [kbld-config.yaml](../../packages/kbld-config.yaml)
+
+2. Start local registry
+
+   Local docker registry is used for pushing the package bundle to get the sha256, to use it later when producing repo bundle.
+
+   Run the below make target to start the local registry
+
+   ```shell
+      make local-registry
+   ```
+
+3. Build the package bundles that belong to a particular package repository
+
+   To build the package bundles for all the packages in the management package repository run:
+
+   ```shell
+      PACKAGE_REPOSITORY=management make push-package-bundles
+   ```
+
+   To build a particular package bundle, run:
+
+   ```shell
+      PACKAGE_REPOSITORY=management PACKAGE_NAME=my-package make package-bundle
+   ```
+
+4. Build package repo bundle
+
+   After building the package bundles, it is time to build package repository bundle, run the below make
+   target:
+
+   ```shell
+      PACKAGE_REPOSITORY=management OCI_REGISTRY=${REGISTRY} make package-repo-bundle
+   ```
+
+5. Push package bundles
+
+   After the package bundles are generated, now it's time to push them to an OCI registry, so that they can be consumed, run
+   the below make target:
+
+   ```shell
+      PACKAGE_REPOSITORY=management OCI_REGISTRY=${REGISTRY} make push-all-package-bundles
+   ```
+
+   If you are interested in pushing only a few package bundles, you could do that by running
+
+   ```shell
+      PACKAGE_REPOSITORY=management OCI_REGISTRY=${REGISTRY} make push-package-bundles ${PACKAGE_BUNDLES}
+   ```
+
+   PACKAGE_BUNDLES variable should be comma-separated values and must not contain spaces.
+   Example: PACKAGE_BUNDLES=featuregates,core-management-plugins
+
+6. Push repo bundle
+
+   To push the generated repo bundle to an OCI registry, run:
+
+   ```shell
+      PACKAGE_REPOSITORY=management OCI_REGISTRY=quay.io/hyayiv make push-package-repo-bundle
+   ```
+
+Follow the below steps to test the artifacts that are generated in previous steps on your local cluster
+
+We can use the `package` plugin to add the repo bundle to the cluster and to install the packages
+
+1. Add package repo bundle to the cluster
```

Comment:
Does this need a `tanzu login` to the cluster before running `package` commands? If so, a note about that would be helpful.